### PR TITLE
Update to use minimum_dependencies

### PR DESCRIPTION
**Pull request by nabobalis**
Brings it in line with the template. 

Also checks we are working with our min deps before our next major release.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -42,51 +42,33 @@ setenv =
...
+    # Handle minimum dependencies via minimum_dependencies
+    oldestdeps: minimum_dependencies
     # Figure tests need a tightly controlled environment
     figure-!devdeps: astropy==6.0.0
     figure-!devdeps: matplotlib==3.8.2
     figure-!devdeps: mpl-animators==1.1.1
 extras =
     tests
 commands_pre =
+    oldestdeps: minimum_dependencies sunkit_image --filename requirements-min.txt
```
Comment (nabobalis) [inline tox.ini:70]:
```suggestion
    oldestdeps: minimum_dependencies sunpy --filename requirements-min.txt
```

**Review by alasdairwilson — APPROVED** [review]

**Comment by nabobalis** [comment]
Hmm, it doesnt seem to be working.

File: pyproject.toml [hunk 2]
```diff
@@ -29,8 +29,8 @@ authors = [
 ]
 dependencies = [
   "astropy>=5.2.0",
-  "numpy>=1.23.0",
-  "packaging>=21.3",
+  "numpy>=1.23.5",
+  "packaging>=23.0",
```
Comment (nabobalis) [inline pyproject.toml:33]: I might have to set this manually in the tox instead of a bump.

File: pyproject.toml [hunk 3]
```diff
@@ -94,7 +94,10 @@ timeseries = [
   "matplotlib>=3.5.0",
   "pandas>=1.4.0",
 ]
-visualization = ["matplotlib>=3.5.0", "mpl-animators>=1.0.0"]
+visualization = [
+  "matplotlib>=3.5.0",
+  "mpl-animators>=1.0.0"
```
Comment (nabobalis) [inline pyproject.toml:99]:
```suggestion
  "mpl-animators>=1.0.0",
```

**Pull request merged**